benchmarks/utils/evaluation.py

```diff
@@ -53,6 +53,29 @@ def _get_completed_instances(self) -> set[EvalInstanceID]:
             )
         return completed_instances
 
+    def _prepare_instances_for_resume(self) -> List[EvalInstance]:
+        """Prepare instances for resume case without completed filtering.
+
+        This ensures we get the full set of selected instances (from --select)
+        so we can properly intersect with failed instances from previous attempts.
+        """
+        from benchmarks.utils.dataset import get_dataset
+
+        df = get_dataset(
+            dataset_name=self.metadata.dataset,
+            split=self.metadata.dataset_split,
+            eval_limit=self.metadata.eval_limit,
+            completed_instances=None,  # Don't filter completed instances
+            selected_instances_file=self.metadata.selected_instances_file,
+        )
+
+        instances: List[EvalInstance] = []
+        for _, row in df.iterrows():
+            inst_id = str(row["instance_id"])
+            instances.append(EvalInstance(id=inst_id, data=row.to_dict()))
+
+        return instances
```

Comment:
I don't think it is right to define a function like this since the assumption here is that child class will implement `prepare_instances` to return the instance for evaluation data.

We should not make assumption about how child class will load data: here we made the assumption that they generate data by doing `get_dataset` which might not be valid all the time.